lib/ansible/modules/cloud/azure/azure_rm_applicationsecuritygroup_facts.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2019 Yunge Zhu, <yungez@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_applicationsecuritygroup_facts
+version_added: "2.8"
+short_description: Get Azure Application Security Group facts.
+description:
+    - Get facts of Azure Application Security Group.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group.
+    name:
+        description:
+            - The name of the application security group.
+    tags:
+        description:
+            - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Yunge Zhu (@yungezz)"
+
+'''
+
+EXAMPLES = '''
+  - name: List application security group in specific resource group
+    azure_rm_applicationsecuritygroup_facts:
+      resource_group: myResourceGroup
+
+  - name: List application security group in specific subscription
+    azure_rm_applicationsecuritygroup_facts:
+
+  - name: Get application security group by name
+    azure_rm_applicationsecuritygroup_facts:
+        resource_group: myResourceGroup
+        name: myApplicationSecurityGroup
+        tags:
+            - foo
+'''
+
+RETURN = '''
+applicationsecuritygroups:
+    description:
+        - List of application security groups.
+    returned: always
+    type: complex
+    contains:
+        id:
+            description: Id of the application security group.
+            type: str
+            returned: always
+            sample:
+                "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Network/applicationSecurityGroups/MyAsg"
+        location:
+            description:
+                - Loation of the application security group.
+            type: str
+            returned: always
+            sample: eastus
+        name:
+            description:
+                - Name of the resource.
+            type: str
+            returned: always
+            sample: myAsg
+        provisioning_state:
+            description:
+                - Provisioning state of application security group.
+            type: str
+            returned: always
+            sample: Succeeded
+        resource_guid:
+            description:
+                - The resource GUID property of the application security group resource.
```

Comment:
I am not really sure if we shuld have guid returned.
I think there's no real use for the user

Reply:
this one is not same with id. seems it's specific to asg, so i choose to return it